### DISCUSSION TITLE
Screen wake up event

### DISCUSSION THREAD

**Discussion by zdunex25** [zepp-health/zeppos-forum#76] [Q&A]
@MyoungXUE I started to play with animations recently and users feedback is that they like animations, but not when they play everytime you interact with your watch. For now if I wanna start animation on _raise to wake_ gesture I have to call it in **resume_call**, but **resume_call** gets called everytime user goes to the main screen, from setting, notifications etc. Do you know anything about how to recognize raise to wake gesture?

[animation preview](https://i.postimg.cc/8k9t3MkM/ezgif-com-video-to-gif.gif)

**Comment by htoooth** [comment]
hi @zdunex25, 
This feature is currently not supported, and it's not easy to implement either. This is because the watch face startup time occurs before the screen power-up time.

**Reply by zdunex25** [reply]
According to this:
![image](https://user-images.githubusercontent.com/1674725/219345489-d78ba0ca-24e7-4540-bf83-1af9850fdf44.png)
I tried to control animation start by boolean, I pass true in resume_call and false in pause_call yet this does not work until watchface reloads, so I have to select another WF then go back to mine WF and only then boolean status it properly read.
If raise to wake implementation is problematic, please add support for **hmApp.reloadPage(option)** for watchfaces in code, so we could call it and enjoy @htoooth

**Reply by htoooth** [reply]
hi @zdunex25 

`resume_call` and `pause_call` methods do not support parameters. The watch face is loaded when it is first set. Therefore, hm.reloadPage is not useful. Do you want to reload in resume_call? This also doesn't work, as it would cause the watch face to be repeatedly loaded.

**Reply by zdunex25** [reply]
I saw redraw() as well, I just wish to have a method that will reload a watchface on button press so it could properly read boolean status.
